src/lib.rs

```diff
@@ -57,8 +61,32 @@ macro_rules! pyo3_built {
         target.set_item("endianness", $info::CFG_ENDIAN)?;
         target.set_item("pointer-width", $info::CFG_POINTER_WIDTH)?;
         target.set_item("profile", $info::PROFILE)?;
-        info.set_item("target", target)?;
+        $dict.set_item("target", target)?;
+    };
+    ($py: ident, $info: ident, $dict: ident, "git") => {
+        let git = PyDict::new($py);
+        git.set_item("version", $info::GIT_VERSION)?;
+        git.set_item("dirty", $info::GIT_DIRTY)?;
+        git.set_item("hash", $info::GIT_COMMIT_HASH)?;
+        git.set_item("head", $info::GIT_HEAD_REF)?;
+        $dict.set_item("git", git)?;
+
+    };
+    ($py: ident, $info: ident, $dict: ident, "finish") => {
+        $dict
+    };
+    // Default build
+    ($py: ident, $info: ident) => {
+        pyo3_built!{$py, $info, "build", "time", "deps", "features", "host", "target"}
+    };
+    // Custom build
+    ($py: ident, $info: ident, $($i:tt ),+ ) => {{
```

Comment:
Can you check if making `$i` a literal instead of a token-tree here would work? This could help debugging.